src/dataflow/components/tools/dataflow-tool.tsx

```diff
@@ -82,11 +83,15 @@ export default class DataflowToolComponent extends BaseComponent<IProps, IState>
             programContent.setProgramStartEndTime(startTime, endTime);
           }
         });
-
+        const properties: IOtherDocumentProperties = { dfRunId: id };
+        if (datasetName.length > 0) {
+          const dataProp = "dfHasData";
+          properties[dataProp] = "true";
```

Comment:
```suggestion
          properties.dfHasData = "true";
```
minor simplification